src/gsi/m_nc_berror.f90

```diff
@@ -0,0 +1,908 @@
+  rc=0; mype_=0; root_=0
+  if(present(myid) .and. present(root) ) then
+     mype_ = myid
+     root_ = root
+  endif
+ 
+! Open the file. NF90_NOWRITE tells netCDF we want read-only access to
+! the file.
+  call check_( nf90_open(fname, NF90_NOWRITE, ncid), rc, mype_, root_ )
+  if(rc/=0) return
+
+! Read global attributes
+  call check_( nf90_inq_dimid(ncid, "lon", varid), rc, mype_, root_)
+  call check_( nf90_inquire_dimension(ncid, varid, len=nlon), rc, mype_, root_ )
+  call check_( nf90_inq_dimid(ncid, "lat", varid), rc, mype_, root_ )
+  call check_( nf90_inquire_dimension(ncid, varid, len=nlat), rc, mype_, root_ )
+  call check_( nf90_inq_dimid(ncid, "lev", varid), rc, mype_, root_ )
+  call check_( nf90_inquire_dimension(ncid, varid, len=nlev), rc, mype_, root_ )
+
+! Close the file, freeing all resources.
+  call check_( nf90_close(ncid), rc, mype_, root_ )
+
+  return
+
+end subroutine read_dims_
+
+subroutine read_berror_ (fname,bvars,rc, myid,root)
+  implicit none
+  character(len=*), intent(in)    :: fname ! input filename
+  type(nc_berror_vars),intent(inout) :: bvars ! background error variables
+  integer, intent(out) :: rc
+  integer, intent(in), optional :: myid,root ! accommodate MPI calling programs
+
+! This will be the netCDF ID for the file and data variable.
+  integer :: ncid, varid
+
+! Local variables
+  character(len=*), parameter :: myname_ = myname//"::read_"
+  character(len=4) :: cindx
+  integer :: nv,nl,nlat,nlon,nlev
+  integer :: ndims_, nvars_, ngatts_, unlimdimid_
+  integer :: nlat_,nlon_,nlev_
+  integer :: mype_,root_
+  real(4), allocatable :: data_in(:,:,:)
+  logical :: verbose
+  logical :: init_
+  
+   
+! Return code (status)
+  rc=0; mype_=0; root_=0
+  verbose=.true.
+  init_=.false.
+  if(present(myid).and.present(root) )then
+    if(myid/=root) verbose=.false.
+    mype_ = myid
+    root_ = root
+  endif
+ 
+! Get dimensions
+  call read_dims_ (fname,nlat_,nlon_,nlev_,rc, mype_,root_)
+
+  init_ = bvars%initialized
+  if ( init_ ) then
+!   Set dims
+    nlat=bvars%nlat
+    nlon=bvars%nlon
+    nlev=bvars%nsig
+
+!   Consistency check
+    if (nlon_ /= nlon .or. nlat_ /=nlat .or. nlev_/=nlev ) then
+       rc=1
+       if(myid==root) then
+         print *, 'nlat(file) = ', nlat_, 'nlat(required) = ', nlat
+         print *, 'nlon(file) = ', nlon_, 'nlon(required) = ', nlon
+         print *, 'nlev(file) = ', nlev_, 'nlev(required) = ', nlev
+         print *, myname_,  'Inconsistent dimensions, aborting ... '
+       endif
+       return
+    endif
+  else
+!   Set dims
+    nlat=nlat_
+    nlon=nlon_
+    nlev=nlev_
+    call init_berror_vars_(bvars,nlon,nlat,nlev)
+  endif
+
+! Open the file. NF90_NOWRITE tells netCDF we want read-only access to
+! the file.
+
+  call check_( nf90_open(fname, NF90_NOWRITE, ncid), rc, mype_, root_ )
+  if(rc/=0) return
+
+! Read global attributes
+! call check_( nf90_inquire(ncid, ndims_, nvars_, ngatts_, unlimdimid_), rc, mype_, root_ )
```

Comment:
no need for these global attributes?

Reply:
Remove commented lines.  Done at 68d52ac